### DISCUSSION TITLE
Detecting the language field using AI when registering a DOI in DataCite Fabrica

### DISCUSSION THREAD

**Discussion by paulvierkant** [datacite/datacite-suggestions#123] [General Suggestions]
### What is the problem that your suggestion solves?

I manually have to add the language type for the following fields:
1. titles
2. description

### What solution might meet your needs?

Use an AI tool to detect the language in the language field of the DataCite Fabrica form. Pre-fill it but just as a suggestion that can still be changed by the users. This saves approx. 3 seconds when filling out the form

### Your name

Paul Vierkant

### Your organization

DataCite

### What group(s) would benefit from your suggestion?

- [X] DataCite members registering DOIs and metadata and administering accounts
- [ ] Developers building integrations with DataCite services
- [ ] Harvesters who index DataCite DOIs and metadata
- [ ] Individuals or institutions analyzing DataCite DOIs and metadata
- [X] DataCite staff
- [ ] Other

**Comment by adambuttrick** [comment]
We did quite a bit of work around language detection in ROR for a harder problem (detecting for org name metadata) with the code and results [here](https://github.com/adambuttrick/language-detection-tests). There are a couple off-the-shelf libraries ([fastText](https://fasttext.cc/docs/en/language-identification.html) and [lingua](https://github.com/pemistahl/lingua-py)) that can be used at a fairly high level of precision:

https://github.com/adambuttrick/language-detection-tests

**Reply by paulvierkant** [reply]
Thanks. And even if the detection is wrong, this should be just a suggestion that can be changed by the creator.
